lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -123,7 +121,7 @@ data NetworkLayer m block = NetworkLayer
         -- ^ Get the slot corresponding to a cursor.
 
     , currentNodeTip
-        :: ExceptT ErrCurrentNodeTip m BlockHeader
```

Comment:
Nice. Can't remember: Is the current `currentNodeTip` throwing in IO or blocking if the socket is not there?

Reply:
It's just reading a `TVar`. If the socket is not there, you'll get the intial value - genesis.